packages/client/src/object/convertWireToOsdkObjects.ts

```diff
@@ -112,14 +115,53 @@ const protoConverterCache = createCache(
 
 const isAfterFeb2024OrNewApis = false;
 
-/**
- * @param objs the objects to be converted, the contents of this array will be mutated
+/*
+ * Currently loses all information about the interface. Future PRs will improve
+ * FIXME
  */
-export async function convertWireToOsdkObjectsInPlace(
+export async function convertWireToOsdkInterfaceInPlace(
   client: MinimalClient,
-  objs: OntologyObjectV2[],
+  objects: OntologyObjectV2[],
+  interfaceApiName: string,
 ) {
-  // Fix properties
+  fixObjectPropertiesInline(objects);
+
+  const localObjectCache = createAsyncCache((client, apiName: string) =>
+    client.ontology.provider.getObjectDefinition(apiName)
+  );
+
+  const uniqueApiNames = new Set<string>(objects.map(o => o.$apiName));
```

Comment:
`objects` may be large enough that `for (const {$apiName} of objects) { uniqueApiNames.add($apiName); }` is notably faster by avoiding the .map step. you'd have to jsbench it to be sure.